### PR TITLE
adding `systems` to flake

### PR DESCRIPTION
**Pull request by TornaxO7**
This allows our flake to be used for arm and intel + amd cpus.

### CODE REVIEW

**Comment by eneoli** [comment]
@TornaxO7, I don't quite understand. Didn't it work with at least Intel and AMD before as well? What exactly does nix-systems/nix-systems do? I don't see quite the difference from how it was before.

**Comment by TornaxO7** [comment]
I've got another idea, but this will make use of [flake-parts](https://flake.parts/). I'll explain there, what this improves.

**Comment by TornaxO7** [comment]
Alright, after testing it a bit out, I think [flake-parts](https://ferrous-systems.com/blog/officially-qualified-ferrocene/) is a bit overkill here, so I'd like to jump back to [nix-systems] or even close this tbh.

Basically, [nix-systems] is an organization which holds a list with different architecture groups. You can see [here](https://github.com/nix-systems) the different "profiles": All `system`s which are only for linux, for darwin (apple) or riscv-linux etc.

With this, we wouldn't need to manage an array of `systems` on our own.

## TL;DR
[nix-systems] replaces our
```nix
forAllSystems = function:
        nixpkgs.lib.genAttrs [
          "x86_64-linux"
        ]
          (system: function (import nixpkgs {
            inherit system;

            overlays = [ rust-overlay.overlays.default ];
          }));
```
and we don't have to keep that up to date since "they do it for us".

So yeah, technically it's doesn't give us that much benefits, but I think it's neat. I'll give you the final decision, if you want to have this or not.

[nix-systems]: https://github.com/nix-systems/nix-systems

**Review by eneoli — COMMENTED** [review]
What exactly would there be to "keep up to date"?

**Comment by TornaxO7** [comment]
> What exactly would there be to "keep up to date"?

[This list](https://github.com/eneoli/flakeshot/blob/main/flake.nix#L10-L12).

**Comment by eneoli** [comment]
How likely is it that there will be changes that affect Flakeshot?

**Comment by TornaxO7** [comment]
> How likely is it that there will be changes that affect Flakeshot?

Hm... very unlikely I think

**Comment by eneoli** [comment]
Okay, I close this then as I don’t see improvements but only new dependencies.

If I missed something or you have another reason for using this, feel free to reopen.
